components/script/layout_interface.rs

```diff
@@ -95,6 +95,8 @@ pub trait LayoutRPC {
     fn content_box(&self) -> ContentBoxResponse;
     /// Requests the dimensions of all the content boxes, as in the `getClientRects()` call.
     fn content_boxes(&self) -> ContentBoxesResponse;
+    /// Requests the clientTop.
+    fn client_top(&self) -> ClientGeometryResponse;
```

Comment:
Maybe we should group this into a single `get_geometry()` umbrella function with an associated `GeometryMetric` enum containing `ClientTop`, `ClientWidth` enum so we don't have a dozen functions here. (BTW, `ClientGeometry` probably isn't the right word, since there are non-"client" metrics too; `offsetTop`, for example.)